### PR TITLE
Stabilize app, add database layer and terraform

### PR DESCRIPTION
**Pull request by nbpeth**
**Description** 
* simplify messages between client and server
* error handling
* patch memory leak in websocket subject
* dashboard and active session components create their own connections
* add database 
* terraform for provisioning RDS
* various UI/UX tweaks

### CODE REVIEW

**Comment by yemble** [comment]
Recommend renaming the database to something like `storypoints-release` (to distinguish from test/dev without using prod (since it's internal). Also declare the db user/pass vars in tf.

**Comment by yemble** [comment]
Is it mongo or mysql or postgresql? I see all three.

**Comment by nbpeth** [comment]
I don't see any postgres references anywhere. I'm removing the mongo client and npm deps

**Comment by yemble** [comment]
The terraform has postgres as the engine. You could probably just change those to mysql 👍

File: ops/terraform/main.tf [hunk 1]
```diff
@@ -5,3 +5,43 @@ resource "aws_ecr_repository" "rp-story-points" {
...
+  engine_version = "5.7"
+  identifier = "example"
+  instance_class = "db.t2.micro"
+  maintenance_window = "Sun:05:54-Sun:06:24"
+  monitoring_role_arn = "${data.aws_iam_role.rds_monitoring_role.arn}"
+  monitoring_interval = "60"
+  multi_az = false
+  name = "rpstorypointsdb"
+  option_group_name = "default:postgres-10"
+  parameter_group_name = "default.postgres10"
```
Comment (nbpeth) [inline ops/terraform/main.tf]: so weak - code search didn't show this for `postgres`

**Comment by mars64** [comment]
Had to fix a few things in the terraform code, manually planned and applied: 

```
Terraform will perform the following actions:

  # aws_db_instance.release will be created
  + resource "aws_db_instance" "release" {
      + address                               = (known after apply)
      + allocated_storage                     = 10
      + allow_major_version_upgrade           = false
      + apply_immediately                     = (known after apply)
      + arn                                   = (known after apply)
      + auto_minor_version_upgrade            = false
      + availability_zone                     = (known after apply)
      + backup_retention_period               = 10
      + backup_window                         = "06:40-07:10"
      + ca_cert_identifier                    = (known after apply)
      + character_set_name                    = (known after apply)
      + copy_tags_to_snapshot                 = true
      + db_subnet_group_name                  = "vpc-test-private"
      + deletion_protection                   = true
      + endpoint                              = (known after apply)
      + engine                                = "mysql"
      + engine_version                        = "5.7"
      + hosted_zone_id                        = (known after apply)
      + id                                    = (known after apply)
      + identifier                            = "storypointsrelease"
      + identifier_prefix                     = (known after apply)
      + instance_class                        = "db.t2.small"
      + kms_key_id                            = (known after apply)
      + license_model                         = (known after apply)
      + maintenance_window                    = "sun:05:54-sun:06:24"
      + max_allocated_storage                 = 20
      + monitoring_interval                   = 60
      + monitoring_role_arn                   = "arn:aws:iam::596297932419:role/rds-monitoring-role"
      + multi_az                              = false
      + name                                  = "storypointsrelease"
      + option_group_name                     = "default:mysql-5-7"
      + parameter_group_name                  = "default.mysql5.7"
      + password                              = (sensitive value)
      + performance_insights_enabled          = false
      + performance_insights_kms_key_id       = (known after apply)
      + performance_insights_retention_period = (known after apply)
      + port                                  = (known after apply)
      + publicly_accessible                   = false
      + replicas                              = (known after apply)
      + resource_id                           = (known after apply)
      + skip_final_snapshot                   = false
      + status                                = (known after apply)
      + storage_encrypted                     = true
      + storage_type                          = "gp2"
      + tags                                  = {
          + "Name"        = "storypoints-release"
          + "application" = "story-points"
          + "contact"     = "parrtnerdevelopment@validity.com"
          + "environment" = "test"
          + "managedby"   = "terraform"
          + "project"     = "story-points"
          + "repo_name"   = "ReturnPath/story-points/"
          + "team"        = "partner development"
        }
      + timezone                              = (known after apply)
      + username                              = "storypoints_release"
      + vpc_security_group_ids                = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

**Comment by nbpeth** [comment]
thanks @mars64 !

**Pull request closed without merging**